main.go

```diff
@@ -202,22 +225,34 @@ func buildParseErrorString(target, e string, err error) string {
 	return msg
 }
 
+func SetupDeferredAccessLogging(accessLogger *zap.Logger, accessLogDetails *carbonapipb.AccessLogDetails, t time.Time, logAsError bool)  {
```

Comment:
same here. It's only used in 'main'. Can be unexported. You still can access private functions from *_test files.